File: app/src/main/java/com/afriserve/smsmanager/data/csv/CsvPreviewService.java
package com.afriserve.smsmanager.data.csv;

import android.util.Log;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 * Preview and validation service for CSV data
 */
public class CsvPreviewService {
    private static final String TAG = "CsvPreviewService";
    private static final int PREVIEW_COUNT = 3;

    /**
     * Generate message preview for first N recipients with data substitution
     */
    public static List<String> generateMessagePreviews(
            String template,
            List<CsvRecipient> recipients,
            int count
    ) {
        List<String> previews = new ArrayList<>();
        int actualCount = Math.min(count, recipients.size());

        for (int i = 0; i < actualCount; i++) {
            CsvRecipient recipient = recipients.get(i);
            String previewMessage = template;

            // Replace all placeholders with actual values
            if (recipient.getData() != null) {
                for (Map.Entry<String, String> entry : recipient.getData().entrySet()) {
                    String placeholder = "{" + entry.getKey().toLowerCase().replace(" ", "") + "}";
                    previewMessage = previewMessage.replace(placeholder, entry.getValue());
                }
            }

            // Add phone number to preview
            String preview = previewMessage + "\n\n[To: " + recipient.getPhoneNumber() + "]";
            previews.add(preview);
        }

        return previews;
    }

    /**
     * Validate phone numbers and return summary
     */
    public static PhoneValidationResult validatePhoneNumbers(List<CsvRecipient> recipients) {
        List<String> validNumbers = new ArrayList<>();
        List<String> invalidNumbers = new ArrayList<>();

        for (CsvRecipient recipient : recipients) {
            if (isValidPhoneNumber(recipient.getPhoneNumber())) {
                validNumbers.add(recipient.getPhoneNumber());
            } else {
                invalidNumbers.add(recipient.getPhoneNumber());
            }
        }

        return new PhoneValidationResult(
                recipients.size(),
                validNumbers.size(),
                invalidNumbers.size(),
                validNumbers,
                invalidNumbers
        );
    }

    /**
     * Create summary message for upload result
     */
    public static String createSummary(
            CsvParsingResult parsingResult,
            String fileName
    ) {
        StringBuilder summary = new StringBuilder();
        summary.append("📊 CSV Upload Summary\n");
        summary.append("━".repeat(40)).append("\n");
        summary.append("File: ").append(fileName).append("\n");
        summary.append("\n");
        summary.append("Recipients Detected: ").append(parsingResult.getTotalCount()).append("\n");
        summary.append("✓ Valid Recipients: ").append(parsingResult.getValidCount()).append("\n");
        summary.append("✗ Invalid Recipients: ").append(parsingResult.getInvalidCount()).append("\n");
        summary.append("\n");
        summary.append("Columns Available: ").append(parsingResult.getColumnCount()).append("\n");
        
        if (parsingResult.getDetectionResult() != null) {
            summary.append("Phone Column: ").append(
                parsingResult.getDetectionResult().getDetectedPhoneColumnName() != null ? 
                parsingResult.getDetectionResult().getDetectedPhoneColumnName() : "Not detected"
            ).append("\n");
        }
        
        summary.append("\n");
        
        if (parsingResult.getDetectionResult() != null && 
            parsingResult.getDetectionResult().getPotentialPhoneColumns() != null &&
            parsingResult.getDetectionResult().getPotentialPhoneColumns().size() > 1) {
            summary.append("⚠️  Multiple phone columns detected:\n");
            for (String column : parsingResult.getDetectionResult().getPotentialPhoneColumns()) {
                summary.append("  • ").append(column).append("\n");
            }
        }
        
        if (parsingResult.getParseErrors() != null && !parsingResult.getParseErrors().isEmpty()) {
            summary.append("\n");
            summary.append("⚠️  Parsing errors:\n");
            for (String error : parsingResult.getParseErrors()) {
                summary.append("  • ").append(error).append("\n");
            }
        }

        return summary.toString();
    }

    /**
     * Simple phone number validation
     */
    private static boolean isValidPhoneNumber(String phoneNumber) {
        if (phoneNumber == null || phoneNumber.trim().isEmpty()) {
            return false;
        }

        String digitsOnly = phoneNumber
                .replace("+", "")
                .replace(" ", "")
                .replace("-", "")
                .replace("(", "")
                .replace(")", "")
                .replace(".", "")
                .replace(",", "");

        return digitsOnly.length() >= 10 && digitsOnly.length() <= 15 && digitsOnly.matches("\\d+");
    }
}

File: app/src/tmp_src/java/com/afriserve/smsmanager/data/csv/CsvPreviewService.kt
package com.afriserve.smsmanager.data.csv

import android.util.Log

/**
 * Preview and validation service for CSV data
 */
class CsvPreviewService {
    companion object {
        private const val TAG = "CsvPreviewService"
        private const val PREVIEW_COUNT = 3
        
        /**
         * Generate message preview for first N recipients with data substitution
         */
        fun generateMessagePreviews(
            template: String,
            recipients: List<CsvRecipient>,
            count: Int = PREVIEW_COUNT
        ): List<String> {
            val previews = mutableListOf<String>()
            
            recipients.take(count).forEach { recipient ->
                var previewMessage = template
                
                // Replace all placeholders with actual values
                recipient.data.forEach { (key, value) ->
                    val placeholder = "{${key.lowercase().replace(" ", "")}}"
                    previewMessage = previewMessage.replace(placeholder, value, ignoreCase = true)
                }
                
                // Add phone number to preview
                val preview = "$previewMessage\n\n[To: ${recipient.phoneNumber}]"
                previews.add(preview)
            }
            
            return previews
        }
        
        /**
         * Validate phone numbers and return summary
         */
        fun validatePhoneNumbers(recipients: List<CsvRecipient>): PhoneValidationResult {
            val validNumbers = mutableListOf<String>()
            val invalidNumbers = mutableListOf<Pair<String, String>>()
            
            recipients.forEach { recipient ->
                if (isValidPhoneNumber(recipient.phoneNumber)) {
                    validNumbers.add(recipient.phoneNumber)
                } else {
                    invalidNumbers.add(recipient.phoneNumber to "Invalid format")
                }
            }
            
            return PhoneValidationResult(
                totalCount = recipients.size,
                validCount = validNumbers.size,
                invalidCount = invalidNumbers.size,
                validPhoneNumbers = validNumbers,
                invalidPhoneNumbers = invalidNumbers
            )
        }
        
        /**
         * Create summary message for upload result
         */
        fun createSummary(
            parsingResult: CsvParsingResult,
            fileName: String
        ): String {
            return buildString {
                appendLine("📊 CSV Upload Summary")
                appendLine("━".repeat(40))
                appendLine("File: $fileName")
                appendLine()
                appendLine("Recipients Detected: ${parsingResult.totalCount}")
                appendLine("✓ Valid Recipients: ${parsingResult.validCount}")
                appendLine("✗ Invalid Recipients: ${parsingResult.invalidCount}")
                appendLine()
                appendLine("Columns Available: ${parsingResult.columnCount}")
                if (parsingResult.detectionResult != null) {
                    appendLine("Phone Column: ${parsingResult.detectionResult.detectedPhoneColumnName ?: "Not detected"}")
                }
                appendLine()
                if (parsingResult.detectionResult?.potentialPhoneColumns?.size ?: 0 > 1) {
                    appendLine("⚠️  Multiple phone columns detected:")
                    parsingResult.detectionResult?.potentialPhoneColumns?.forEach {
                        appendLine("  • $it")
                    }
                }
                if (parsingResult.parseErrors.isNotEmpty()) {
                    appendLine()
                    appendLine("⚠️  Parsing errors:")
                    parsingResult.parseErrors.forEach {
                        appendLine("  • $it")
                    }
                }
            }
        }
        
        /**
         * Simple phone number validation
         */
        private fun isValidPhoneNumber(phoneNumber: String): Boolean {
            if (phoneNumber.isBlank()) return false
            
            val digitsOnly = phoneNumber
                .replace("+", "")
                .replace(" ", "")
                .replace("-", "")
                .replace("(", "")
                .replace(")", "")
                .replace(".", "")
            
            return digitsOnly.length in 10..15 && digitsOnly.all { it.isDigit() }
        }
    }
}

/**
 * Result of phone number validation
 */
data class PhoneValidationResult(
    val totalCount: Int,
    val validCount: Int,
    val invalidCount: Int,
    val validPhoneNumbers: List<String>,
    val invalidPhoneNumbers: List<Pair<String, String>>
) {
    val validityPercentage: Int
        get() = if (totalCount > 0) (validCount * 100) / totalCount else 0
}
